batch/batch/cloud/terra/azure/driver/driver.py

```diff
@@ -0,0 +1,554 @@
+    @staticmethod
+    async def create(default_region: str) -> 'SingleRegionMonitor':
+        return SingleRegionMonitor(default_region)
+
+    def __init__(self, default_region: str):
+        self._default_region = default_region
+
+    def default_location(self) -> str:
+        return self._default_region
+
+    def choose_location(
+        self,
+        cores: int,  # pylint: disable=unused-argument
+        local_ssd_data_disk: bool,  # pylint: disable=unused-argument
+        data_disk_size_gb: int,  # pylint: disable=unused-argument
+        preemptible: bool,  # pylint: disable=unused-argument
+        regions: List[str],
+    ) -> str:
+        return self._default_region
+
+
+def create_disk_config(
+    disk_name: str,
+    disk_resource_id: str,
+    disk_size_gb: int,
+) -> Dict[str, Any]:
+    return {
+        'common': {
+            'name': disk_name,
+            'description': disk_name,
+            'cloningInstructions': 'COPY_NOTHING',
+            'accessScope': 'PRIVATE_ACCESS',
+            'managedBy': 'USER',
+            'resourceId': disk_resource_id,
+            'properties': [],
+        },
+        'azureDisk': {
+            'name': disk_name,
+            'size': disk_size_gb,
+        },
+    }
+
+
+def create_vm_config(
+    file_store: FileStore,
+    location: str,
+    machine_name: str,
+    machine_type: str,
+    activation_token: str,
+    max_idle_time_msecs: int,
+    instance_config: InstanceConfig,
+):
+    BATCH_WORKER_IMAGE = os.environ['HAIL_BATCH_WORKER_IMAGE']
+    HAIL_DOMAIN = os.environ['HAIL_DOMAIN']
+    TERRA_STORAGE_ACCOUNT = os.environ['TERRA_STORAGE_ACCOUNT']
+    WORKSPACE_STORAGE_CONTAINER_ID = os.environ['WORKSPACE_STORAGE_CONTAINER_ID']
+    WORKSPACE_STORAGE_CONTAINER_URL = os.environ['WORKSPACE_STORAGE_CONTAINER_URL']
+    WORKSPACE_MANAGER_URL = os.environ['WORKSPACE_MANAGER_URL']
+    WORKSPACE_ID = os.environ['WORKSPACE_ID']
+
+    instance_config_base64 = base64.b64encode(json.dumps(instance_config.to_dict()).encode()).decode()
+
+    assert isinstance(deploy_config, TerraDeployConfig)
+    assert isinstance(instance_config, TerraAzureSlimInstanceConfig)
+
+    startup_script = fr'''#cloud-config
+
+mounts:
+  - [ ephemeral0, null ]
+  - [ ephemeral0.1, null ]
+
+write_files:
+  - owner: batch-worker:batch-worker
+    path: /startup.sh
+    content: |
+      #!/bin/bash
+
+      set -ex
+
+      function cleanup() {{
+          set +x
+          sleep 1000
+          token=$(az account get-access-token --query accessToken --output tsv)
+
+          VM_RESOURCE_ID={ instance_config._resource_id }
+          curl -X POST "{ WORKSPACE_MANAGER_URL }/api/workspaces/v1/$WORKSPACE_ID/resources/controlled/azure/vm/$VM_RESOURCE_ID" \
+              -H  "accept: */*" \
+              -H  "Authorization: Bearer $token" \
+              -H  "Content-Type: application/json" \
+              -d "{{\"jobControl\":{{\"id\":\"$VM_RESOURCE_ID\"}}}}"
```

Comment:
presumably this deletes the vm but I'm surprised Therese no -XDELETE or `/delete` or `"action":"delete"`?

Reply:
Apparently this is a restriction imposed by what why use to generate their [APIs](https://workspace.dsde-dev.broadinstitute.org/swagger-ui.html#/ControlledAzureResource/deleteAzureVm)

Reply:
> Delete a controlled Azure VM resource. This is async, because the operation can take a few minutes. OpenAPI does not support request body in DELETE, but async state requires it. Hence this is a POST.

Reply:
Heh. I really hate that GET XXX gets metadata about the resource but POST XXX deletes it. That just feels dangerous, but, it is what it is.